packages/core/src/search/search.ts

```diff
@@ -203,6 +203,11 @@ function parseKeyValue(searchRequest: SearchRequest, key: string, value: string)
     modifier = '';
   }
 
+  if (code === '_has' || key.includes('.')) {
```

Comment:
Since this is a special case where we're deferring parsing, I'd rather not get it intertwined with normal search parsing — makes it simpler to understand what's going on in my opinion